### PR TITLE
doq 0.7.0 (new formula)

### PR DESCRIPTION
**Pull request by graelo**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

### CODE REVIEW

File: Formula/doq.rb [hunk 1]
```diff
@@ -0,0 +1,57 @@
+class Doq < Formula
+  include Language::Python::Virtualenv
+
+  desc "Docstring generator for Python"
+  homepage "https://github.com/heavenshell/py-doq"
+  url "https://files.pythonhosted.org/packages/35/a0/c22615949f5270e8a0819d5a1278330468eeb6f8bfe495f574c5a869b57e/doq-0.7.0.tar.gz"
+  sha256 "c323dc59857a1dbea3e7d71e73c6fb11a88a565c3a225b75e446f70f0659049b"
+  license "BSD-3-Clause"
+
+  depends_on "python"
```
Comment (nandahkrishna) [inline Formula/doq.rb:10]:
```suggestion
  depends_on "python@3.9"
```

File: Formula/doq.rb [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+  def install
+    virtualenv_install_with_resources
+  end
+
+  test do
+    ENV["LC_ALL"] = "en_US.UTF-8"
+
+    # Test that the docstring can be generated correctly for the following
+    # source code
```
Comment (nandahkrishna) [inline Formula/doq.rb:35]:
```suggestion
```

File: Formula/doq.rb [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    source_code = <<~EOS
+      def compute(arg1, arg2: str) -> str:
+          pass
+    EOS
+    (testpath/"compute.py").write(source_code)
+
+    # Expected code with generated docstring
```
Comment (nandahkrishna) [inline Formula/doq.rb:42]:
```suggestion
```

**Review by nandahkrishna — COMMENTED** [review]
Thanks for your PR, @u0xy. Please see my suggestions below.

**Comment by u0xy** [comment]
The tests are failing because

```
GitHub repository not notable enough (<30 forks, <30 watchers and <75 stars)
```

May I suggest this is made clear in the Formula Cookbook? There is no mention of that requirement. Is this a true blocker?

While I'm at it, I think the text color of the Formula Cookbook makes the text not very legible ;)

**Comment by nandahkrishna** [comment]
> May I suggest this is made clear in the Formula Cookbook? There is no mention of that requirement. Is this a true blocker?

For what it's worth, https://docs.brew.sh/Formula-Cookbook#audit-the-formula mentions running `brew audit --new-formula foo`, which would have spotted the earlier `depends_on "python"` issue and also the notability issue:
```
➜ brew audit --new-formula doq
doq:
  * Dependency 'python' is an alias; use the canonical name 'python@3.9'.
  * GitHub repository not notable enough (<30 forks, <30 watchers and <75 stars)
Error: 2 problems in 1 formula detected
```
If you think it should be made explicit, please submit a pull request and we'll happily review it.

This is unfortunately a blocker, but you can maintain the formula in a personal tap if you wish to. When it meets the requirements for inclusion into homebrew-core, you can resubmit the formula. Here are some docs that could help you get started:

https://docs.brew.sh/Taps
https://docs.brew.sh/Interesting-Taps-and-Forks
https://docs.brew.sh/How-to-Create-and-Maintain-a-Tap
https://brew.sh/2020/11/18/homebrew-tap-with-bottles-uploaded-to-github-releases/

**Comment by u0xy** [comment]
Thanks! In fact, I did not get this warning when running `brew audit --new --strict doq` locally, but maybe this is because the formula was already installed from my tap. Sorry for the noise.

**Pull request closed without merging**